src/Constraint/Constraint.php

```diff
@@ -55,10 +55,13 @@ class Constraint extends AbstractConstraint
     );
 
     /** @var string */
-    private $operator;
+    protected $operator;
```

Comment:
IMO, it should stay private. Turning things protected means you have to take them into account for BC, so it should not be done without a use case making this the right extension point (and I don't see a reason for that)